From 411d2019a1d333396f6d33b667390947b045aeae Mon Sep 17 00:00:00 2001
From: Jasper Orschulko <Jasper.Orschulko@iris-sensing.com>
Date: Wed, 2 Jun 2021 02:09:55 +0200
Subject: [PATCH] Add static and dynamic lib targets to Makefile

Upstream-Status: Submitted [https://github.com/cesanta/mongoose/pull/1295]
---
 Makefile | 41 ++++++++++++++++++++++++++++++++++++++++-
 1 file changed, 39 insertions(+), 0 deletion(-)

diff --git a/Makefile b/Makefile
index a5c25cad..97bb441a 100644
--- a/Makefile
+++ b/Makefile
@@ -14,6 +14,16 @@ IPV6 ?= 1
 ASAN_OPTIONS ?=
 EXAMPLES := $(wildcard examples/*)
 EXAMPLE_TARGET ?= example
+SO_MAJOR = 7
+SO_MINOR = 2
+LIBNAME = libmongoose
+SHARED_LIB = $(LIBNAME).so.$(SO_MAJOR).$(SO_MINOR)
+STATIC_LIB = $(LIBNAME).a
+LIB_OBJ = mongoose.o
+PREFIX ?= /usr/local
+LIBDIR ?= $(PREFIX)/lib
+INCLUDEDIR ?= $(PREFIX)/include
+DOCDIR ?= $(PREFIX)/share/doc
 .PHONY: ex test
 
 ifeq "$(SSL)" "MBEDTLS"
@@ -95,6 +105,35 @@ linux: Makefile mongoose.c mongoose.h test/unit_test.c
 linux++: CC = g++ -Wno-missing-field-initializers
 linux++: linux
 
+linux-libs: $(LIBNAME).so.$(SO_MAJOR).$(SO_MINOR) $(LIBNAME).a
+
+$(SHARED_LIB): $(LIB_OBJ)
+	$(CC) $(LIB_OBJ) $(LDFLAGS) -shared -o $(SHARED_LIB)
+
+$(STATIC_LIB):
+	$(AR) rcs $(STATIC_LIB) $(LIB_OBJ)
+
+install: linux-libs
+	install -Dm644 $(STATIC_LIB) $(DESTDIR)$(LIBDIR)/$(STATIC_LIB)
+	install -Dm644 $(SHARED_LIB) $(DESTDIR)$(LIBDIR)/$(SHARED_LIB)
+	install -Dm644 mongoose.h $(DESTDIR)$(INCLUDEDIR)/mongoose.h
+	ln -s $(SHARED_LIB) $(DESTDIR)$(LIBDIR)/$(LIBNAME).so.$(SO_MAJOR)
+	ln -s $(SHARED_LIB) $(DESTDIR)$(LIBDIR)/$(LIBNAME).so
+	install -dm755 $(DESTDIR)$(DOCDIR)/mongoose
+	cp -a examples $(DESTDIR)$(DOCDIR)/mongoose/
+	cp -a docs/* $(DESTDIR)$(DOCDIR)/mongoose/
+
+uninstall:
+	rm -f $(STATIC_LIB) $(DESTDIR)$(LIBDIR)/$(STATIC_LIB)
+	rm -f $(SHARED_LIB) $(DESTDIR)$(LIBDIR)/$(SHARED_LIB)
+	rm -f $(DESTDIR)$(INCLUDEDIR)/mongoose.h
+	rm -f $(DESTDIR)$(LIBDIR)/$(LIBNAME).so.$(SO_MAJOR)
+	rm -f $(DESTDIR)$(LIBDIR)/$(LIBNAME).so
+	rm -rf $(DESTDIR)$(DOCDIR)/mongoose
+
+arm: Makefile mongoose.c mongoose.h test/unit_test.c
+	$(ARM) arm-none-eabi-gcc mongoose.c -c -Itest -DMG_ARCH=MG_ARCH_CUSTOM $(OPTS) $(WARN) $(INCS) -DMG_MAX_HTTP_HEADERS=5 -DMG_ENABLE_LINES -DMG_ENABLE_DIRECTORY_LISTING=0 -DMG_ENABLE_SSI=1
+
 mongoose.c: $(SRCS) Makefile
 	(cat src/license.h; echo; echo '#include "mongoose.h"' ; (for F in src/private.h src/*.c ; do echo; echo '#ifdef MG_ENABLE_LINES'; echo "#line 1 \"$$F\""; echo '#endif'; cat $$F | sed -e 's,#include ".*,,'; done))> $@
 
-- 
2.31.1

